api/server/http-rest/options.go

```diff
@@ -0,0 +1,47 @@
+package http_rest
+
+import (
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Option func(g *Server) error
+
+func WithMuxHandler(m restHandler) Option {
+	return func(g *Server) error {
+		g.cfg.muxHandler = m
+		return nil
+	}
+}
+
+func WithHTTPAddr(addr string) Option {
+	return func(g *Server) error {
+		g.cfg.httpAddr = addr
+		return nil
+	}
+}
+
+// WithRouter allows adding a custom mux router to the gateway.
```

Comment:
need to update this, removing gateway wording, same with comments below

Reply:
Currently I am of the opinion that self-explanatory function names don't need to be documented. A simple `// WithRouter --` suffices to get no warnings.